astropy/io/fits/tests/test_image.py

```diff
@@ -1777,6 +1778,37 @@ def test_compressed_header_missing_znaxis(self):
         with pytest.raises(TypeError):
             comp_hdu.compressed_data
 
+    @pytest.mark.parametrize(('keyword', 'dtype', 'expected'),
+                             [(kk, dd, ee) for kk, (dd, ee) in
+                             itertools.product(["BSCALE", "BZERO"],
```

Comment:
These are only 8 cases currently. I agree that `product` and `zip` are nice but I had to run this in the interpreter to work out what it evaluated to. Maybe that could be spelled out explicitly?

```
    @pytest.mark.parametrize(
        ('keyword', 'dtype', 'expected'),
        [('BSCALE', (numpy.uint8, numpy.float32), ('BSCALE', (numpy.int16, numpy.float32),
         ('BSCALE', (numpy.int32, numpy.float64), ('BZERO', (numpy.uint8, numpy.float32),
         ('BZERO', (numpy.int16, numpy.float32), ('BZERO', numpy.int32, numpy.float64)]
```